### PR TITLE
unfurling: RPCs for setting and getting settings intended for UI CORE-9399

### PR DESCRIPTION
**Pull request by mmaxim**
Also fixes a bug in `TestChatSrvUnfurl` intro'd by the favicon changes.

### CODE REVIEW

File: go/chat/server.go [hunk 1]
```diff
@@ -2346,3 +2346,41 @@ func (h *Server) ResolveUnfurlPrompt(ctx context.Context, arg chat1.ResolveUnfur
...
+	uid, err := utils.AssertLoggedInUID(ctx, h.G())
+	if err != nil {
+		return res, err
+	}
+	settings, err := h.G().Unfurler.GetSettings(ctx, uid)
+	if err != nil {
+		return res, err
+	}
+	res.Mode = settings.Mode
+	for w := range settings.Whitelist {
```
Comment (joshblum) [inline go/chat/server.go]: why not `copy`?

Comment (mmaxim) [reply go/chat/server.go]: Source is a `map`

**Comment by mmaxim** [comment]
Merging over CI

File: go/chat/server.go (hunk 1 above)
Comment (joshblum) [reply go/chat/server.go]: ah makes sense

**Pull request merged**